_assets/ci/Jenkinsfile.tests-anvil

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env groovy
+library 'status-jenkins-lib@v1.9.6'
+
+pipeline {
+  agent { label 'linux && x86_64 && nix-2.19' }
+
+  parameters {
+    string(
+      name: 'BRANCH',
+      defaultValue: 'develop',
+      description: 'Name of branch to build.'
+    )
+  }
+
+  options {
+    timestamps()
+    /* Prevent Jenkins jobs from running forever */
+    timeout(time: 30, unit: 'MINUTES')
+    disableConcurrentBuilds()
+    /* manage how many builds we keep */
+    buildDiscarder(logRotator(
+      numToKeepStr: '5',
+      daysToKeepStr: '30',
+      artifactNumToKeepStr: '1',
+    ))
+  }
+
+  environment {
+    PLATFORM = 'tests-anvil'
+    PKG_URL = "${currentBuild.absoluteUrl}/consoleText"
+
+    DB_CONT     = "status-go-test-db-${env.EXECUTOR_NUMBER.toInteger() + 1}"
+    DB_PORT     = "${5432 + env.EXECUTOR_NUMBER.toInteger()}"
+    TMPDIR      = "${WORKSPACE_TMP}"
+    GOPATH      = "${WORKSPACE_TMP}/go"
+    GOCACHE     = "${WORKSPACE_TMP}/gocache"
+    PATH        = "${PATH}:${GOPATH}/bin"
+    REPO_SRC    = "${GOPATH}/src/github.com/status-im/status-go"
+    BASE_BRANCH = "${env.CHANGE_TARGET}"
+  }
+
+  stages {
+    stage('Prep') {
+      steps { /* Go needs to find status-go in GOPATH. */
+        sh "mkdir -p \$(dirname ${REPO_SRC})"
+        sh "ln -s ${WORKSPACE} ${REPO_SRC}"
+      }
+    }
+
+    stage('Vendor Check') {
+      steps { script {
+        nix.shell('make vendor', pure: false)
+        /* fail build if vendoring hasn't been done */
+        nix.shell('git diff --exit-code --no-color --stat vendor/')
+      } }
+    }
+
+    stage('Migration') {
+      when { // https://github.com/status-im/status-go/issues/4993#issuecomment-2022685544
+        expression { !isTestNightlyJob() }
+      }
+      steps { script {
+        nix.shell('make migration-check', pure: false)
+      } }
+    }
+
+    stage('Commit') {
+      environment {
+        BASE_BRANCH = "${env.BASE_BRANCH}"
+      }
+      when { // https://github.com/status-im/status-go/issues/4993#issuecomment-2022685544
+        expression { !isTestNightlyJob() }
+      }
+      steps { script {
+        nix.shell('make commit-check', pure: false)
+      } }
+    }
+
+    stage('Lint') {
+      steps { script {
+        nix.shell('make lint', pure: true)
+      } }
+    }
+
+    stage('Canary') {
+      steps { script {
+        nix.shell('make canary-test', pure: true)
+      } }
+    }
```

Comment:
Why did you guys add the same steps as in `Jenkinsfile.tests`? What's the point of that?